x-pack/plugins/enterprise_search/server/routes/app_search/search_settings.ts

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { schema } from '@kbn/config-schema';
```

Comment:
I am treating these routes as pass through. For that reason, you will see the following 3 things:

- These are still named Search Settings, NOT Relevance Tuning. This is to stay 1 to 1 with the backend search settings API.
- Original routes have not been modified
- I am doing no more than a high level check for keys as validation. I do not want to introduce redundant schema validation, since this is already validated on the server.

Reply:
> I am doing no more than a high level check for keys as validation. I do not want to introduce redundant schema validation, since this is already validated on the server.

I think this is fine, but I think we should come to a consensus on how we want to handle this sort of validation universally

Reply:
I'm fine with this as part of migration speed, but I'd really like to come back and do a cleanup round this post-MVP. There are a lot of issues with our current backend APIs:

- Not RESTful, inconsistent naming (e.g. `queries` vs `query` in the Analytics APIs, and I still don't know what the logic was behind `/detail` and `/collection` routes)
- Inconsistent casing all over the place with both sent and returned payloads, etc.
- Old/outdated naming (Search Settings vs Relevance Tuning, Reference UI vs Search UI, etc. etc.)

That being said hopefully one of our future goal is to move to our public APIs anyway and not rely so much on internal ones, so who knows what the future holds.

Re: validation, ya I think we'd need to deeper dive into that as well whenever we circle back. One thing to potentially think about is that schema errors we catch in Kibana can potentially be i18n'd, whereas errors we get back from the server are always in English. I actually don't know if there's a good solution to that and we may just have to be OK with it, but it's definitely an interesting topic at least